Mathlib/GroupTheory/MonoidLocalization/DivPairs.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała
+-/
+import Mathlib.GroupTheory.MonoidLocalization.Basic
+
+/-!
+# Submonoid of pairs with quotient in a submonoid
+
+This file defines the submonoid of pairs whose quotient lies in a submonoid of the localization.
+-/
+
+variable {M G H : Type*} [CommMonoid M] [CommGroup G] [CommGroup H]
+  {f : (⊤ : Submonoid M).LocalizationMap G} {g : (⊤ : Submonoid M).LocalizationMap H}
+  {s : Submonoid G} {x : M × M}
```

Comment:
Should these be explicit?

Reply:
No, because they are used in an iff. In general, I find it much less confusing only having implicit argument in `variable`